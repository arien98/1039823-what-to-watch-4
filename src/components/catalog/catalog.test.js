import React from "react";
import renderer from "react-test-renderer";
import Catalog from "./catalog";

describe(`catalog renders correctly`, () => {
  test(`it renders correctly`, () => {
    const props = {
      onFilmCardClick: () => {},
      onFilterClick: () => {},
      currentFilter: `Dramas`,
      children: [],
      genres: [`All genres`, `Dramas`, `Horror`],
      filmsData: [{
        id: 2,
        title: `Bohemian Rhapsody`,
        smallPoster: `img/bohemian-rhapsody.jpg`,
        preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
      },
      {
        id: 3,
        title: `Macbeth`,
        smallPoster: `img/macbeth.jpg`,
        preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
      },
      {
        id: 4,
        title: `Aviator`,
        smallPoster: `img/aviator.jpg`,
        preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
      },
      ],
    };

    const tree = renderer
      .create(<Catalog {...props}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
