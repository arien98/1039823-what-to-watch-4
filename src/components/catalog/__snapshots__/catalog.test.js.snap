// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`catalog renders correctly it renders correctly 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        data-filter="All genres"
        href="#"
        onClick={[Function]}
      >
        All genres
      </a>
    </li>
    <li
      className="catalog__genres-item catalog__genres-item--active"
    >
      <a
        className="catalog__genres-link"
        data-filter="Dramas"
        href="#"
        onClick={[Function]}
      >
        Dramas
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        data-filter="Horror"
        href="#"
        onClick={[Function]}
      >
        Horror
      </a>
    </li>
  </ul>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
    >
      <div
        className="small-movie-card__image"
        data-id={2}
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <img
          alt="Bohemian Rhapsody"
          className="small-movie-card__pic"
          height="175"
          src="img/bohemian-rhapsody.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
        onClick={[Function]}
      >
        <a
          className="small-movie-card__link"
          data-id={2}
          href="movie-page.html"
        >
          Bohemian Rhapsody
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <div
        className="small-movie-card__image"
        data-id={3}
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <img
          alt="Macbeth"
          className="small-movie-card__pic"
          height="175"
          src="img/macbeth.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
        onClick={[Function]}
      >
        <a
          className="small-movie-card__link"
          data-id={3}
          href="movie-page.html"
        >
          Macbeth
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <div
        className="small-movie-card__image"
        data-id={4}
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <img
          alt="Aviator"
          className="small-movie-card__pic"
          height="175"
          src="img/aviator.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
        onClick={[Function]}
      >
        <a
          className="small-movie-card__link"
          data-id={4}
          href="movie-page.html"
        >
          Aviator
        </a>
      </h3>
    </article>
  </div>
</section>
`;
